pointerToSlice/pointerToSlice.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package pointerToSlice
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "pointerToSlice",
+	Doc:  "check for usage of pointer to slice in function definitions",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	checkNode := func(expr ast.Expr) {
+		sexpr, ok := expr.(*ast.StarExpr)
+		if !ok {
+			return
+		}
+		if _, ok := sexpr.X.(*ast.ArrayType); ok {
+			pass.Reportf(sexpr.Pos(), "use of pointer to slice in function definition")
```

Comment:
This would mean a pointer to array, not slice correct?

Reply:
I think it would be both, there isn't a `SliceType` as far as I know.